contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -343,6 +343,45 @@ contract VaultAdmin is VaultStorage {
         emit AssetSupported(_asset);
     }
 
+    /**
+     * @notice Remove a supported asset from the Vault
+     * @param _asset Address of asset
+     */
+    function removeAsset(address _asset) external onlyGovernor {
+        require(assets[_asset].isSupported, "Asset not supported");
+        require(
+            IVault(address(this)).checkBalance(_asset) == 0,
+            "Vault still holds asset"
+        );
+
+        uint256 assetsCount = allAssets.length;
+        uint256 assetIndex = assetsCount;
+        for (uint256 i = 0; i < assetsCount; ++i) {
+            if (allAssets[i] == _asset) {
+                assetIndex = i;
+                break;
+            }
+        }
+
+        // Note: If asset is not present in `allAssets`, the following line
+        // will revert with an out-of-bound error. Right now, there's no
+        // reason why an asset would have `Asset.isSupported = true` but
+        // not exist in `allAssets`.
+
+        // Update allAssets array
+        allAssets[assetIndex] = allAssets[assetsCount - 1];
+        allAssets.pop();
+
+        // Reset default strategy
+        assetDefaultStrategies[_asset] = address(0);
+        emit AssetDefaultStrategyUpdated(_asset, address(0));
+
+        // Mark as unsupported
+        assets[_asset].isSupported = false;
```

Comment:
Let's actually clear everything in the asset struct with `delete`, rather than just isSupported. Feels cleaner and will keep us from having weird data around lurking for the future if we ever decided to turn the asset back on.

Reply:
Sounds good. Will make that change

Reply:
Done